src/server/analytic_services/prometheus_reports/noobaa_endpoint_report.js

```diff
@@ -161,6 +161,38 @@ const NOOBAA_ENDPOINT_METRICS = js_utils.deep_freeze([{
                 10000, 20000, 50000,
             ],
         }
+    },
+    {
+        type: 'Gauge',
+        name: 'semaphore_value',
+        configuration: {
+            help: 'Namespace semaphore value',
+            labelNames: ['bucket_name', 'type']
```

Comment:
does it mean we store it per bucket? why?

Reply:
Hi @dannyzaken,
1. Even if `object_io` and `namespace_fs` have the same metrics name, It has a label `type` which will have the value `ns` or `ns_fs` and we can use these metrics to identify the type of metrics.
2. Yes, sampling will happen only when there is an IO. Once IO is done and semaphore values go back to default values metrics will have that default semaphore value
3. Semaphore values will vary only when there is an IO and most of the time it will have the same value, and it will increase the metrics we push to Prometheus. We can always find the average in specific intervals by using Prometheus Query Language. I think pushing average value to metrics creates confusion at the graph level.
Please share your thoughts

Reply:
I added  label `bucket_name` for debugging purposes, When there is an issue with one bucket using this label we can query only that bucket's metrics from Prometheus.